torch/serialization.py

```diff
@@ -291,6 +291,25 @@ def raise_err_msg(patterns, e):
     except (io.UnsupportedOperation, AttributeError) as e:
         raise_err_msg(["seek", "tell"], e)
 
+def _check_dill_version(pickle_module):
+    '''Checks if using dill as the pickle module, and if so, checks if it is the correct version.
+    If dill version is lower than 0.3.1, a ValueError is raised.
+
+    Args:
+        pickle_module: module used for pickling metadata and objects
+
+    '''
+    if pickle_module.__name__ == 'dill':
+        required_dill_version = (0, 3, 1)
+        dill_version = tuple(int(num) for num in pickle_module.__version__.split('.'))
+
+        if dill_version < required_dill_version:
+            raise ValueError(
+                "'torch' supports 'dill' >= %s, but you have 'dill' %s" % (
```

Comment:
It would be good to suggest all the alternatives. "Please upgrade your dill or use `pickle`".